zjit/src/cruby.rs

```diff
@@ -263,6 +263,10 @@ impl ShapeId {
     pub fn is_too_complex(self) -> bool {
         unsafe { rb_jit_shape_too_complex_p(self.0) }
     }
+
+    pub fn is_frozen(self) -> bool {
+        unsafe { rb_zjit_shape_frozen_p(self.0) }// & SHAPE_ID_FL_FROZEN
```

Comment:
Why not import SHAPE_ID_FL_FROZEN directly instead of going through C?

Reply:
I tried but it was not working. I would love to do that instead. The C function is magnificently silly